include/zephyr/net/ieee802154_radio.h

```diff
@@ -757,61 +757,171 @@ enum ieee802154_config_type {
+	 *    scheduling uncertainties with respect to CSL transmitters into
+	 *    account as specified by the standard such that at least the full
+	 *    SHR of a legitimate RX frame is guaranteed to land inside the
+	 *    cannel sample.
+	 *
+	 *    To this avail, the last configured expected RX time plus an
+	 *    integer number of CSL periods SHALL point exactly to the center of
+	 *    each CSL channel sample:
+	 *
+	 *        rxSlotCenterNs_N = last expected RX time
+	 *            + N * (cslPeriod * 10 * PHY-specific symbol period in ns)
+	 *
+	 *    While the configured CSL period is greater than zero, drivers
+	 *    SHOULD validate the center of each RX slot accordingly. If
+	 *    validation fails, drivers SHOULD log the error but SHALL
+	 *    nevertheless accept and schedule the RX slot with a zero success
+	 *    value to work around minor implementation or rounding errors in
+	 *    upper layers.
+	 *
+	 * Configure and start a CSL receiver:
+	 *
+	 *     ENH_ACK_HEADER_IE
+	 *        |
+	 *        | EXPECTED_RX_TIME (end of SFD of a perfectly timed RX
+	 *        |    |              frame in the first channel sample)
+	 *        |    |
+	 *        |    | CSL_PERIOD (>0)            RX_SLOT
+	 *        |    |    |                          |
+	 *        v    v    v                          v
+	 *     -----------------------------------------------[-CSL channel sample ]----+
+	 *                                         ^                                    |
+	 *                                         |                                    |
+	 *                                         +--------------------- loop ---------+
+	 *
+	 * Disable CSL on the receiver:
+	 *
+	 *     CSL_PERIOD (=0)
+	 *        |
+	 *        v
+	 *     ---------------------
+	 *
+	 * Update the CSL period to a new value:
+	 *
+	 *     EXPECTED_RX_TIME (based on updated period)
+	 *        |
+	 *        |  CSL_PERIOD (>0, updated)       RX_SLOT
+	 *        |     |                              |
+	 *        v     v                              v
+	 *     -----------------------------------------------[-CSL channel sample ]----+
+	 *                                         ^                                    |
+	 *                                         |                                    |
+	 *                                         +--------------------- loop ---------+
 	 *
 	 * @note Available in any interface operational state.
-	 *
-	 * @warning This configuration option does not conform to the
-	 * requirements specified in #61227 as it is incompatible with standard
-	 * primitives and may therefore be deprecated in the future.
 	 */
 	IEEE802154_CONFIG_CSL_PERIOD,
 
 	/**
-	 * @brief Configure the next CSL receive window (i.e. "channel sample")
-	 * center, in units of nanoseconds relative to the network subsystem's
-	 * local clock.
+	 * Configure the timepoint at which the next RX frame is expected to
```

Comment:
I might be nitpicking, but is it relevant if it's the timepoint of the _next_ RX frame? In my understanding this value can serve its purpose also if it points to a moment in the past, or to a moment in the future that is not the next RX frame.

Reply:
No, IMO this is not a nitpick but an important observation as you are pointing out a fundamental design trade-off (re-usability vs. specificity).

What you say is true for CSL (and other applications) where frames are expected to arrive at regular intervals - but not in the general case. It is e.g. wrong for TSCH, beacon synchronization when several PANs are being tracked, RIT or ranging applications with "on-demand" blinks (rather than periodic ones).

We cannot ask the standard for help as requirements due to hardware offloading are considered implementation details by the standard.

The trade off we need to resolve is whether we should introduce one specific configuration attribute per protocol, e.g.:
- TSCH: macTsRxOffset + macTsRxWait/2, section 6.5.4.3
- CSL: Rendezvous Time, CSL Phase
- RIT: Time to First Listen
- BE-PANs (including DSME): expected beacon frame transmissions (may be more than one in case a device tracks several PANs) - section 6.5.2
- Ranging (TDoA) - expected arrival time assuming a given blink period

and so on..

Or *alternatively* whether we should have a single low-level (protocol independent) configuration attribute similar to the "scheduled TX" case that can be used generically.

I'm not really sold on the proposed solution - maybe it's a bad idea. I just thought that an "expected RX time" would be generically useful in many contexts as it is what all those protocols really define on a PHY level. I also think it is good to keep the RX and TX APIs as symmetric as possible for simplicity and consistency.

But even if we use the "generic" solution as proposed by myself, we need to clarify that in the CSL case the "expected RX time" can point to any expected RX time in the past or in the future to define an anchor time.

I changed the wording accordingly - see whether you really like it now or whether we should go for the "one config per protocol" solution.

Reply:
> I just thought that an "expected RX time" would be generically useful in many contexts as it is what all those protocols really define on a PHY level.

I agree. I prefer a single, generic attribute over multiple protocol-specific attributes in this case. I read the rephrased doc and I think it explains the flexibility of this attribute well. Thank you for improving it.